src/plugins/dataflow-tool/components/dataflow-program.tsx

```diff
@@ -1083,21 +1098,66 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private countSerialDataNodes(nodes: Node[]){
     // implementing with a "count" of 1 or 0 in case we need to count nodes in future
     let serialNodesCt = 0;
+
+    //sensor will need serial once these particular sensors are chosen
     nodes.forEach((n) => {
       if(n.data.sensor === "emg" || n.data.sensor === "fsr"){
         serialNodesCt++;
       }
+
+      //live output block will alert need for serial
+      // only after connection to another node is made
+      // this allows user to drag a block out and work on program before connecting
+      if (n.name === "Live Output"){
+        if(n.inputs.entries().next().value[1].connections.length > 0){
+          serialNodesCt++;
+        }
+      }
     });
     // constraining all counts to 1 or 0 for now
     if (serialNodesCt > 0){
       this.stores.serialDevice.setSerialNodesCount(1);
     } else {
       this.stores.serialDevice.setSerialNodesCount(0);
     }
+
+    if (serialNodesCt > 0 && !this.stores.serialDevice.hasPort()){
+      this.postSerialModal();
+    }
   }
 
-  private updateNodeChannelInfo = (n: Node) => {
+  private sendDataToSerialDevice(n: Node){
+    if (isFinite(n.data.nodeValue as number)){
+      this.stores.serialDevice.writeToOut(n.data.nodeValue as number);
+    }
+  }
+
+  private postSerialModal(){
+    const lastMsg = localStorage.getItem('last-connect-message');
+
+    let alertMessage = "";
 
+    // np physical connection made
+    if (lastMsg !== "connect" && this.stores.serialDevice.serialNodesCount > 0){
+      alertMessage += "1. Connect the arduino to your computer. \
+        2. Click the ⚡️ button on the upper left, then choose the device at the prompt.";
+    }
+
+    // physical connection has been made but user action needed
+    if (lastMsg === "connect"
+        && !this.stores.serialDevice.hasPort()
+        && this.stores.serialDevice.serialNodesCount > 0
+    ){
+      alertMessage += "Click the ⚡️ button on the upper left, then choose your device at the prompt.";
```

Comment:
Since this is the same as the message on line 1143 I think it would be worth saving it in a variable.